flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/DefaultExecutionGraph.java

```diff
@@ -1038,7 +1039,11 @@ private boolean transitionState(JobStatus current, JobStatus newState, Throwable
                     current,
                     newState,
                     error);
-
+            if (error != null) {
+                LOG.warn(
+                        "Print ExecutionGraph {}",
+                        LogStackUtils.getCallStack(Thread.currentThread().getStackTrace()));
+            }
```

Comment:
It should always be the main thread of the `JobMaster` that calls the `transitionState` method.

Reply:
from the log, we can see the scene where some exception occurred.
i think the PR helps to analyze some exceptions and is especially important for newcomers.